app/views/adhoc-payment/index.njk

```diff
@@ -42,7 +42,7 @@
                   name="payment-amount"
                   data-non-numeric=""
                   type="text"
-                  inputmode="numeric"
+                  inputmode="decimal"
```

Comment:
It seems that on iPhone iOS this will display keyboard that has a decimal separator but the separator is locale-dependent (for example, if your region is set to German, you get a comma). Since we only allow a decimal point as a separator, some users will still not be able to enter pence. So I think removing the `inputmode` altogether and letting it default to text might be the best thing to do for now.

Reply:
Good call

Reply:
`inputmode=text` (default) is a generally worse experience on mobile (varies significantly based on device, potentially two or three keyboard toggles to get the right combo of numbers and symbols you might need). Is it worth the worse user experience for what I assume is a fairly small percentage of a low volume?

I would think making parsing resilient to `,` or `.` wouldn't be too much work. I haven't looked at the code but I'm guessing it's quite manual, using something like https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat might make it more future proof

Reply:
Looking at making the parsing resilient seems worthwhile (and a usability improvement in general) but I have no idea how many other decimal separators are out there that we might need to deal with.

On balance, I think we cannot really block some users from paying altogether in order to provide a better keyboard for most users. The default keyboard is not that terrible for decimal input. iOS did not support `inputmode` at all until iOS 13, so it was always the default keyboard for them until a couple of years ago. And there are many, many forms out there that do not use `inputmode`, so we will not be the only place where users get the default keyboard even though it is not the best.